### ISSUE TITLE
[tune] Hyperparameter Not Finished

### ISSUE THREAD

**Issue by vinhdiesal** [ray-project/ray#9440]
The following screenshot is from one run of the `mnist` dataset using `tensorflow` and `ray`. 

I used the `tune.run` functions with some hyperparameters. 

What I don't understand is why some trainining got cut off before it finished training. 

See below.

![Screenshot_2020-07-13 TensorBoard(1)](https://user-images.githubusercontent.com/47830328/87295092-8a822400-c4b9-11ea-8ee5-997ac3578d44.png)

**Comment by richardliaw** [comment]
Can you post your `tune.run` call?

**Comment by vinhdiesal** [comment]
Here is the section: 
```
%%time
if __name__ == "__main__":
    import ray
    from ray import tune
    from ray.tune.schedulers import AsyncHyperBandScheduler
    mnist.load_data()  # we do this on the driver because it's not threadsafe
    ray.init(address='auto')
    sched = AsyncHyperBandScheduler(
        time_attr="training_iteration",
        metric="mean_accuracy",
        mode="max",
        max_t=400,
        grace_period=20)

    tune.run(
        train_mnist,
        name="exp",
        scheduler=sched,
        stop={
            "mean_accuracy": 0.99,
            "training_iteration": 5 if args.smoke_test else 300
        },
        num_samples=10,
        resources_per_trial={
            "cpu": 10,
            "gpu": 2
        },
        config={
            "threads": 2,
            "lr": tune.sample_from(lambda spec: np.random.uniform(0.001, 0.1)),
            "momentum": tune.sample_from(
                lambda spec: np.random.uniform(0.1, 0.9)),
            "hidden": tune.sample_from(
                lambda spec: np.random.randint(32, 512)),
        })
```

**Comment by vinhdiesal** [comment]
any updates?

**Comment by richardliaw** [comment]
Hi @vinhdiesal please comment out the scheduler; it is used for early terminating of bad trials.

Thanks!

**Comment by vinhdiesal** [comment]
thanks you
